### ISSUE TITLE
Reconstruction system not working anymore

### ISSUE THREAD

**Issue by Fred3D-tech** [intel-isl/Open3D#2061]
**IMPORTANT: Please use the following template to report the bug.**
_____

**Describe the bug**
AttributeError: 'bool' object has no attribute 'lower'

**To Reproduce**
just use run_system.py

**Environment (please complete the following information):**

- Operating system: Windows 10 64-bit
- Python version: Python 3.7
- Open3D version: the last one (just updated it)
- Is this remote workstation:  no
- How did you install Open3D: conda

**Additional context**
making fragments from RGBD sequence.
Traceback (most recent call last):
  File "C:/Anaconda3/Open3D-master/examples/python/ReconstructionSystem/run_system.py", line 68, in <module>
    make_fragments.run(config)
  File "C:\Anaconda3\Open3D-master\examples\python\ReconstructionSystem\make_fragments.py", line 177, in run
    if config["python_multi_threading"].lower() == "true":
AttributeError: 'bool' object has no attribute 'lower'

Process finished with exit code 1

**Comment by Fred3D-tech** [comment]
ok... to solve it, you need to change line 177 to:
    if str(config["python_multi_threading"]).lower() == "true":

**Comment by Fred3D-tech** [comment]
but now, I have another bug:
  File "C:\Anaconda3\Open3D-master\examples\python\ReconstructionSystem\make_fragments.py", line 68, in make_posegraph_for_fragment
    pose_graph = o3d.pipelines.registration.PoseGraph()
AttributeError: module 'open3d' has no attribute 'pipelines'

**Comment by Fred3D-tech** [comment]
I tried to build in from source using this ref : http://www.open3d.org/docs/release/compilation.html
It fails...
1>------ Build started: Project: ZERO_CHECK, Configuration: Release x64 ------
1>Checking Build System
2>------ Build started: Project: EncodeShader, Configuration: Release x64 ------
3>------ Build started: Project: 3rdparty_qhull_r, Configuration: Release x64 ------
4>------ Build started: Project: 3rdparty_glew, Configuration: Release x64 ------
5>------ Build started: Project: glfw, Configuration: Release x64 ------
6>------ Build started: Project: 3rdparty_jsoncpp, Configuration: Release x64 ------
7>------ Build started: Project: 3rdparty_lzf, Configuration: Release x64 ------
8>------ Build started: Project: 3rdparty_rply, Configuration: Release x64 ------
9>------ Build started: Project: 3rdparty_tinyfiledialogs, Configuration: Release x64 ------
2>Building Custom Rule C:/Anaconda3/Open3D/cpp/tools/CMakeLists.txt
2>EncodeShader.cpp
9>Building Custom Rule C:/Anaconda3/Open3D/CMakeLists.txt
9>tinyfiledialogs.c
3>Building Custom Rule C:/Anaconda3/Open3D/CMakeLists.txt
4>Building Custom Rule C:/Anaconda3/Open3D/CMakeLists.txt
5>Building Custom Rule C:/Anaconda3/Open3D/3rdparty/GLFW/src/CMakeLists.txt
6>Building Custom Rule C:/Anaconda3/Open3D/CMakeLists.txt
7>Building Custom Rule C:/Anaconda3/Open3D/CMakeLists.txt
3>global_r.c
5>context.c
3>stat_r.c
4>glew.c
3>geom2_r.c
3>poly2_r.c
3>merge_r.c
3>libqhull_r.c
3>geom_r.c
3>poly_r.c
3>qset_r.c
3>mem_r.c
3>random_r.c
3>usermem_r.c
3>userprintf_r.c
3>io_r.c
3>user_r.c
5>init.c
2>EncodeShader.vcxproj -> C:\Anaconda3\Open3D\build\bin\Release\EncodeShader.exe
3>rboxlib_r.c
3>userprintf_rbox_r.c
6>json_reader.cpp
6>json_value.cpp
6>json_writer.cpp
5>input.c
8>Building Custom Rule C:/Anaconda3/Open3D/CMakeLists.txt
5>monitor.c
9>3rdparty_tinyfiledialogs.vcxproj -> C:\Anaconda3\Open3D\build\lib\Release\Open3D_3rdparty_tinyfiledialogs.lib
5>vulkan.c
10>------ Skipped Build: Project: check-cpp-style, Configuration: Release x64 ------
10>Project not selected to build for this solution configuration 
11>------ Skipped Build: Project: check-style, Configuration: Release x64 ------
11>Project not selected to build for this solution configuration 
12>------ Skipped Build: Project: apply-cpp-style, Configuration: Release x64 ------
12>Project not selected to build for this solution configuration 
13>------ Build started: Project: ShaderFileTarget, Configuration: Release x64 ------
14>------ Skipped Build: Project: apply-style, Configuration: Release x64 ------
14>Project not selected to build for this solution configuration 
15>------ Skipped Build: Project: uninstall, Configuration: Release x64 ------
15>Project not selected to build for this solution configuration 
7>lzf_c.c
7>lzf_d.c
5>window.c
3>3rdparty_qhull_r.vcxproj -> C:\Anaconda3\Open3D\build\lib\Release\Open3D_3rdparty_qhull_r.lib
5>win32_init.c
16>------ Build started: Project: 3rdparty_qhullcpp, Configuration: Release x64 ------
13>Generating ../../../../cpp/open3d/visualization/shader/Shader.h
8>rply.c
5>win32_joystick.c
7>3rdparty_lzf.vcxproj -> C:\Anaconda3\Open3D\build\lib\Release\Open3D_3rdparty_lzf.lib
16>Building Custom Rule C:/Anaconda3/Open3D/CMakeLists.txt
5>win32_monitor.c
8>3rdparty_rply.vcxproj -> C:\Anaconda3\Open3D\build\lib\Release\Open3D_3rdparty_rply.lib
6>3rdparty_jsoncpp.vcxproj -> C:\Anaconda3\Open3D\build\lib\Release\Open3D_3rdparty_jsoncpp.lib
16>Coordinates.cpp
16>PointCoordinates.cpp
16>Qhull.cpp
16>QhullFacet.cpp
5>win32_time.c
16>QhullFacetList.cpp
16>QhullFacetSet.cpp
16>QhullHyperplane.cpp
16>QhullPoint.cpp
5>win32_thread.c
5>win32_window.c
5>wgl_context.c
13>Building Custom Rule C:/Anaconda3/Open3D/cpp/open3d/visualization/CMakeLists.txt
5>egl_context.c
16>QhullPointSet.cpp
16>QhullPoints.cpp
16>QhullQh.cpp
16>QhullRidge.cpp
16>QhullSet.cpp
16>QhullStat.cpp
16>QhullVertex.cpp
16>QhullVertexSet.cpp
5>osmesa_context.c
5>Generating Code...
16>RboxPoints.cpp
16>RoadError.cpp
16>RoadLogEvent.cpp
5>glfw.vcxproj -> C:\Anaconda3\Open3D\build\lib\Release\glfw3.lib
16>3rdparty_qhullcpp.vcxproj -> C:\Anaconda3\Open3D\build\lib\Release\Open3D_3rdparty_qhullcpp.lib
4>3rdparty_glew.vcxproj -> C:\Anaconda3\Open3D\build\lib\Release\Open3D_3rdparty_glew.lib
17>------ Build started: Project: pipelines, Configuration: Release x64 ------
18>------ Build started: Project: io, Configuration: Release x64 ------
19>------ Build started: Project: geometry, Configuration: Release x64 ------
20>------ Build started: Project: core, Configuration: Release x64 ------
21>------ Build started: Project: camera, Configuration: Release x64 ------
22>------ Build started: Project: visualization, Configuration: Release x64 ------
23>------ Build started: Project: utility, Configuration: Release x64 ------
18>Building Custom Rule C:/Anaconda3/Open3D/cpp/open3d/io/CMakeLists.txt
23>Building Custom Rule C:/Anaconda3/Open3D/cpp/open3d/utility/CMakeLists.txt
21>Building Custom Rule C:/Anaconda3/Open3D/cpp/open3d/camera/CMakeLists.txt
22>Building Custom Rule C:/Anaconda3/Open3D/cpp/open3d/visualization/CMakeLists.txt
20>Building Custom Rule C:/Anaconda3/Open3D/cpp/open3d/core/CMakeLists.txt
21>PinholeCameraIntrinsic.cpp
18>FeatureIO.cpp
21>PinholeCameraParameters.cpp
18>FileFormatIO.cpp
21>PinholeCameraTrajectory.cpp
18>IJsonConvertibleIO.cpp
18>ImageIO.cpp
18>ImageWarpingFieldIO.cpp
18>LineSetIO.cpp
18>OctreeIO.cpp
18>PinholeCameraTrajectoryIO.cpp
22>GeometryRenderer.cpp
22>ImageMaskShader.cpp
22>ImageShader.cpp
22>NormalShader.cpp
22>PhongShader.cpp
22>PickingShader.cpp
22>RGBDImageShader.cpp
22>ShaderWrapper.cpp
23>Console.cpp
20>AdvancedIndexing.cpp
20>ShapeUtil.cpp
20>CUDAUtils.cpp
20>Indexer.cpp
23>Eigen.cpp
23>FileSystem.cpp
20>MemoryManager.cpp
23>Helper.cpp
18>PointCloudIO.cpp
23>IJsonConvertible.cpp
20>MemoryManagerCPU.cpp
20>Tensor.cpp
23>Timer.cpp
20>TensorKey.cpp
18>PoseGraphIO.cpp
22>Simple2DShader.cpp
22>SimpleBlackShader.cpp
18>TriangleMeshIO.cpp
18>VoxelGridIO.cpp
18>FileBIN.cpp
18>FileGLTF.cpp
20>TensorList.cpp
18>FileJPG.cpp
18>FileJSON.cpp
22>SimpleShader.cpp
22>TexturePhongShader.cpp
22>TextureSimpleShader.cpp
20>IndexGetSet.cpp
20>IndexGetSetCPU.cpp
22>ColorMap.cpp
20>NonZero.cpp
18>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\shared\basetsd.h(77,29): error C2371: 'INT32': redefinition; different basic types (compiling source file C:\Anaconda3\Open3D\cpp\open3d\io\file_format\FileJPG.cpp)
20>NonZeroCPU.cpp
20>UnaryEW.cpp
18>C:\Anaconda3\Library\include\jmorecfg.h(216): message : see declaration of 'INT32' (compiling source file C:\Anaconda3\Open3D\cpp\open3d\io\file_format\FileJPG.cpp)
20>UnaryEWCPU.cpp
22>DrawGeometry.cpp
18>FileLOG.cpp
20>BinaryEW.cpp
18>FileOBJ.cpp
21>camera.vcxproj -> C:\Anaconda3\Open3D\build\cpp\open3d\camera\camera.dir\Release\camera.lib
22>GLHelper.cpp
18>FileOFF.cpp
18>FilePCD.cpp
18>FilePLY.cpp
18>FilePNG.cpp
22>PointCloudPicker.cpp
22>SelectionPolygon.cpp
22>SelectionPolygonVolume.cpp
20>BinaryEWCPU.cpp
20>Reduction.cpp
20>ReductionCPU.cpp
18>FilePTS.cpp
22>RenderOption.cpp
18>FileSTL.cpp
22>RenderOptionWithEditing.cpp
22>ViewControl.cpp
18>FileTUM.cpp
22>ViewControlWithCustomAnimation.cpp
22>ViewControlWithEditing.cpp
18>FileXYZ.cpp
22>ViewParameters.cpp
18>FileXYZN.cpp
22>ViewTrajectory.cpp
18>FileXYZRGB.cpp
22>Visualizer.cpp
22>VisualizerCallback.cpp
22>VisualizerRender.cpp
22>VisualizerWithCustomAnimation.cpp
22>VisualizerWithEditing.cpp
22>VisualizerWithKeyCallback.cpp
23>utility.vcxproj -> C:\Anaconda3\Open3D\build\cpp\open3d\utility\utility.dir\Release\utility.lib
20>core.vcxproj -> C:\Anaconda3\Open3D\build\cpp\open3d\core\core.dir\Release\core.lib
18>Done building project "io.vcxproj" -- FAILED.
22>VisualizerWithVertexSelection.cpp
22>visualization.vcxproj -> C:\Anaconda3\Open3D\build\cpp\open3d\visualization\visualization.dir\Release\visualization.lib
24>------ Build started: Project: Open3D, Configuration: Release x64 ------
24>Building Custom Rule C:/Anaconda3/Open3D/cpp/open3d/CMakeLists.txt
24>Open3DConfig.cpp
24>LINK : fatal error LNK1181: cannot open input file 'C:\Anaconda3\Open3D\build\cpp\open3d\io\io.dir\Release\FileJPG.obj'
24>Done building project "Open3D.vcxproj" -- FAILED.
25>------ Build started: Project: pybind, Configuration: Release x64 ------
26>------ Build started: Project: MergeMesh, Configuration: Release x64 ------
27>------ Build started: Project: OdometryRGBD, Configuration: Release x64 ------
28>------ Build started: Project: Voxelization, Configuration: Release x64 ------
29>------ Build started: Project: Visualizer, Configuration: Release x64 ------
30>------ Build started: Project: ViewPCDMatch, Configuration: Release x64 ------
31>------ Build started: Project: TrimMeshBasedOnPointCloud, Configuration: Release x64 ------
32>------ Build started: Project: CameraPoseTrajectory, Configuration: Release x64 ------
26>Building Custom Rule C:/Anaconda3/Open3D/cpp/tools/CMakeLists.txt
30>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
28>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
31>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
29>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
25>Building Custom Rule C:/Anaconda3/Open3D/cpp/pybind/CMakeLists.txt
27>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
30>ViewPCDMatch.cpp
28>Voxelization.cpp
26>MergeMesh.cpp
31>TrimMeshBasedOnPointCloud.cpp
29>Visualizer.cpp
25>camera.cpp
25>blob.cpp
25>core.cpp
27>OdometryRGBD.cpp
25>cuda_utils.cpp
25>device.cpp
25>dtype.cpp
25>size_vector.cpp
25>tensor.cpp
32>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
25>tensor_key.cpp
32>CameraPoseTrajectory.cpp
28>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
31>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
26>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
31>Done building project "TrimMeshBasedOnPointCloud.vcxproj" -- FAILED.
28>Done building project "Voxelization.vcxproj" -- FAILED.
33>------ Build started: Project: ViewGeometry, Configuration: Release x64 ------
34>------ Build started: Project: ConvertPointCloud, Configuration: Release x64 ------
26>Done building project "MergeMesh.vcxproj" -- FAILED.
35>------ Build started: Project: DepthCapture, Configuration: Release x64 ------
25>tensorlist.cpp
25>docstring.cpp
30>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
30>Done building project "ViewPCDMatch.vcxproj" -- FAILED.
36>------ Build started: Project: EvaluateFeatureMatch, Configuration: Release x64 ------
33>Building Custom Rule C:/Anaconda3/Open3D/cpp/tools/CMakeLists.txt
35>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
29>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
27>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
27>Done building project "OdometryRGBD.vcxproj" -- FAILED.
37>------ Build started: Project: ViewDistances, Configuration: Release x64 ------
29>Done building project "Visualizer.vcxproj" -- FAILED.
38>------ Build started: Project: FileDialog, Configuration: Release x64 ------
25>boundingvolume.cpp
25>geometry.cpp
34>Building Custom Rule C:/Anaconda3/Open3D/cpp/tools/CMakeLists.txt
25>halfedgetrianglemesh.cpp
36>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
33>ViewGeometry.cpp
25>image.cpp
38>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
35>DepthCapture.cpp
38>FileDialog.cpp
36>EvaluateFeatureMatch.cpp
34>ConvertPointCloud.cpp
25>kdtreeflann.cpp
37>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
25>lineset.cpp
37>ViewDistances.cpp
25>meshbase.cpp
25>octree.cpp
32>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
32>Done building project "CameraPoseTrajectory.vcxproj" -- FAILED.
39>------ Build started: Project: FileSystem, Configuration: Release x64 ------
35>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
35>Done building project "DepthCapture.vcxproj" -- FAILED.
40>------ Build started: Project: Flann, Configuration: Release x64 ------
33>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
33>Done building project "ViewGeometry.vcxproj" -- FAILED.
41>------ Build started: Project: GLInfo, Configuration: Release x64 ------
34>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
34>Done building project "ConvertPointCloud.vcxproj" -- FAILED.
42>------ Build started: Project: OpenMP, Configuration: Release x64 ------
38>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
38>Done building project "FileDialog.vcxproj" -- FAILED.
43>------ Build started: Project: PCDFileFormat, Configuration: Release x64 ------
41>Building Custom Rule C:/Anaconda3/Open3D/cpp/tools/CMakeLists.txt
25>pointcloud.cpp
25>tetramesh.cpp
40>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
42>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
25>trianglemesh.cpp
41>GLInfo.cpp
40>Flann.cpp
25>voxelgrid.cpp
36>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
36>Done building project "EvaluateFeatureMatch.vcxproj" -- FAILED.
44>------ Build started: Project: PointCloud, Configuration: Release x64 ------
42>OpenMP.cpp
25>class_io.cpp
25>io.cpp
44>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
44>PointCloud.cpp
25>open3d_pybind.cpp
37>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
37>Done building project "ViewDistances.vcxproj" -- FAILED.
45>------ Build started: Project: PoseGraph, Configuration: Release x64 ------
46>------ Build started: Project: ProgramOptions, Configuration: Release x64 ------
46>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
25>color_map.cpp
25>integration.cpp
40>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
40>Done building project "Flann.vcxproj" -- FAILED.
47>------ Build started: Project: RGBDOdometry, Configuration: Release x64 ------
25>odometry.cpp
46>ProgramOptions.cpp
41>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
41>Done building project "GLInfo.vcxproj" -- FAILED.
48>------ Build started: Project: RegistrationRANSAC, Configuration: Release x64 ------
47>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
25>pipelines.cpp
48>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
44>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
44>Done building project "PointCloud.vcxproj" -- FAILED.
49>------ Build started: Project: Image, Configuration: Release x64 ------
50>------ Build started: Project: TriangleMesh, Configuration: Release x64 ------
47>RGBDOdometry.cpp
25>feature.cpp
25>global_optimization.cpp
49>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
48>RegistrationRANSAC.cpp
25>registration.cpp
25>pybind_utils.cpp
49>Image.cpp
25>console.cpp
46>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
46>Done building project "ProgramOptions.vcxproj" -- FAILED.
51>------ Build started: Project: ManuallyCropGeometry, Configuration: Release x64 ------
52>------ Build started: Project: ManuallyAlignPointCloud, Configuration: Release x64 ------
25>eigen.cpp
25>renderoption.cpp
25>viewcontrol.cpp
47>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
47>Done building project "RGBDOdometry.vcxproj" -- FAILED.
53>------ Build started: Project: Log, Configuration: Release x64 ------
52>Building Custom Rule C:/Anaconda3/Open3D/cpp/tools/ManuallyAlignPointCloud/CMakeLists.txt
42>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
42>Done building project "OpenMP.vcxproj" -- FAILED.
54>------ Build started: Project: ColorMapOptimization, Configuration: Release x64 ------
25>visualization.cpp
48>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
48>Done building project "RegistrationRANSAC.vcxproj" -- FAILED.
55>------ Build started: Project: LineSet, Configuration: Release x64 ------
54>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
52>AlignmentSession.cpp
52>ManuallyAlignPointCloud.cpp
52>VisualizerForAlignment.cpp
25>visualizer.cpp
49>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
49>Done building project "Image.vcxproj" -- FAILED.
56>------ Build started: Project: IntegrateRGBD, Configuration: Release x64 ------
55>Building Custom Rule C:/Anaconda3/Open3D/examples/cpp/CMakeLists.txt
54>ColorMapOptimization.cpp
55>LineSet.cpp
57>------ Build started: Project: EvaluatePCDMatch, Configuration: Release x64 ------
52>LINK : fatal error LNK1181: cannot open input file '..\..\..\lib\Release\Open3D.lib'
52>Done building project "ManuallyAlignPointCloud.vcxproj" -- FAILED.
54>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
54>Done building project "ColorMapOptimization.vcxproj" -- FAILED.
55>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
55>Done building project "LineSet.vcxproj" -- FAILED.
25>utility.cpp
25>utility.cpp
25>LINK : fatal error LNK1181: cannot open input file '..\..\lib\Release\Open3D.lib'
25>Done building project "pybind.vcxproj" -- FAILED.
58>------ Build started: Project: python-package, Configuration: Release x64 ------
59>------ Build started: Project: ALL_BUILD, Configuration: Release x64 ------
59>Building Custom Rule C:/Anaconda3/Open3D/CMakeLists.txt
60>------ Skipped Build: Project: INSTALL, Configuration: Release x64 ------
60>Project not selected to build for this solution configuration 
58>CMake Error at C:/Anaconda3/Open3D/cpp/pybind/make_python_package.cmake:20 (file):
58>  file COPY cannot find
58>  "C:/Anaconda3/Open3D/build/lib/Release/Python/pybind.cp37-win_amd64.pyd":
58>  No such file or directory.
58>
58>
58>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(231,5): error MSB6006: "cmd.exe" exited with code 1.
58>Done building project "python-package.vcxproj" -- FAILED.
61>------ Build started: Project: pip-package, Configuration: Release x64 ------
62>------ Skipped Build: Project: conda-package, Configuration: Release x64 ------
62>Project not selected to build for this solution configuration 
63>------ Skipped Build: Project: install-python-package, Configuration: Release x64 ------
63>Project not selected to build for this solution configuration 
61>running bdist_wheel
61>running build
61>running build_py
61>creating build
61>creating build\lib
61>creating build\lib\open3d
61>copying open3d\core.py -> build\lib\open3d
61>copying open3d\j_visualizer.py -> build\lib\open3d
61>copying open3d\__init__.py -> build\lib\open3d
61>running egg_info
61>C:\Anaconda3\lib\site-packages\setuptools\dist.py:485: UserWarning: The version specified ('@PROJECT_VERSION@') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
61>  "details." % self.metadata.version
61>CUSTOMBUILD : error : Invalid distribution name or version syntax: -PYPI-PACKAGE-NAME---PROJECT-VERSION-
61>Done building project "pip-package.vcxproj" -- FAILED.
64>------ Skipped Build: Project: install-pip-package, Configuration: Release x64 ------
64>Project not selected to build for this solution configuration 
========== Build: 18 succeeded, 37 failed, 0 up-to-date, 9 skipped ==========

**Comment by MyEvolution** [comment]
I Got the same issue.

**Comment by sxy2996534527** [comment]
The problem is simply due to a mismatching between the installed version and the latest version of tutorial, just search all the scripts in the Reconstruction system, change "o3d.pipelines" to "o3d". Other small problems are due to the standard difference of the file system, just make a little change accordingly. Well, all work fine finally

**Comment by Fred3D-tech** [comment]
> The problem is simply due to a mismatching between the installed version and the latest version of tutorial, just search all the scripts in the Reconstruction system, change "o3d.pipelines" to "o3d". Other small problems are due to the standard difference of the file system, just make a little change accordingly. Well, all work fine finally
the installed version is the last one.. 
"Other small problems are due to the standard difference of the file system, just make a little change accordingly"
are you serious ? the documentation and "the standard difference of the file system" should be treated in documentation.
If the documentation is wrong and tutorials unusable how can we work ? be serious !

**Comment by sxy2996534527** [comment]
> > The problem is simply due to a mismatching between the installed version and the latest version of tutorial, just search all the scripts in the Reconstruction system, change "o3d.pipelines" to "o3d". Other small problems are due to the standard difference of the file system, just make a little change accordingly. Well, all work fine finally
> > the installed version is the last one..
> > "Other small problems are due to the standard difference of the file system, just make a little change accordingly"
> > are you serious ? the documentation and "the standard difference of the file system" should be treated in documentation.
> > If the documentation is wrong and tutorials unusable how can we work ? be serious !

check your installed version, it's probably 0.10.1, compare the python API of v0.10.1 and of vlatest, there is no "o3d.pipelines" except in vlatest that we can not installed (the latest installed version is not equal to the latest version in doc). Other small problems refer to "import" problems, change it accordingly to your own system. Well, at least, check it and have a try

**Comment by NomitRwt** [comment]
> The problem is simply due to a mismatching between the installed version and the latest version of tutorial, just search all the scripts in the Reconstruction system, change "o3d.pipelines" to "o3d". Other small problems are due to the standard difference of the file system, just make a little change accordingly. Well, all work fine finally
With the proposed change of "o3d.pipelines" to "o3d" the process seems to be working in my environment. Thanks!

**Comment by Fred3D-tech** [comment]
the worst problem is that this project has not been properly maintained for the last 6 months.. with several people I worked with, we participated in this project and some of the code published is part of our work (@baptiste-mnh) ... but today, issues are just not solved, you get no answer and the development and documentation does not help you.. those people don't care.. it is not serious!

**Comment by germanros1987** [comment]
@Fred3D-tech have you finished venting or do you need more time? Should we enable a special space for you to vent? 

In the meantime, while you vent, we are working on adding support for GPU acceleration for Open3D, a new machine learning module, a new rendering engine, a new GUI, a brand new reconstruction system, etc. 

We do care. However, we don't have infinite bandwidth to fix everything on the spot. Please, instead of venting consider helping us maintain the code.

**Comment by masynthetic** [comment]
Can confirm my very fresh install works after changing all instances of 
```
if config["python_multi_threading"].lower() == "true":
```
in the ReconstructionSystem folder to 

```
if str(config["python_multi_threading"]).lower() == "true":
```

and removed all instances of usage of

```
pipelines.
```
(at least 45 of them across all files)

**Comment by pwais** [comment]
+1 `o3d.pipelines` -> `o3d` fixes a lot of code (e.g. the jupyter notebooks)

**Comment by keithahern** [comment]
I also had to change parameters in optimize_posegraph.py

"o3d" was passed as a parameter and was taking precedence over the o3d imports causing errors like:

AttributeError: 'str' object has no attribute 'utility'

renaming the attribute to o3d_ fixed it and the code ran successfully

**Comment by avgilias** [comment]
> The problem is simply due to a mismatching between the installed version and the latest version of tutorial, just search all the scripts in the Reconstruction system, change "o3d.pipelines" to "o3d". Other small problems are due to the standard difference of the file system, just make a little change accordingly. Well, all work fine finally

Excuse me, is possible to explain me where exactly is the Rexonstruction system. I am new in python and programming too.

**Comment by avgilias** [comment]
> The problem is simply due to a mismatching between the installed version and the latest version of tutorial, just search all the scripts in the Reconstruction system, change "o3d.pipelines" to "o3d". Other small problems are due to the standard difference of the file system, just make a little change accordingly. Well, all work fine finally

Excuse me is it possible to explain me a little bit more how to do it and where to find the reconstruction system? I am new in programming

**Comment by theNded** [comment]
The issue has been fixed in #2567.
